src/topology/builder.rs

```diff
@@ -218,41 +219,53 @@ pub async fn build_pieces(
             tracing::Span::none(),
         )
         .unwrap();
-        let mut input_rx = crate::utilization::wrap(Pin::new(input_rx));
 
         let task = match transform {
             Transform::Function(mut t) => {
-                let (output, control) = Fanout::new();
+                let (mut output, control) = Fanout::new();
 
-                let transform = input_rx
+                let mut input_rx = input_rx
                     .filter(move |event| ready(filter_event_type(event, input_type)))
-                    .ready_chunks(128) // 128 is an arbitrary, smallish constant
-                    .inspect(|events| {
+                    .ready_chunks(128); // 128 is an arbitrary, smallish constant
+
+                let mut timer = crate::utilization::Timer::new();
+                let mut last_report = Instant::now();
+
+                let transform = async move {
+                    timer.start_wait();
+                    while let Some(events) = input_rx.next().await {
+                        timer.stop_wait();
+                        if last_report.elapsed().as_secs() >= 5 {
+                            timer.report();
+                            last_report = Instant::now();
+                        }
+
                         emit!(&EventsReceived {
                             count: events.len(),
                             byte_size: events.size_of(),
                         });
-                    })
-                    .flat_map(move |events| {
-                        let mut output = Vec::with_capacity(events.len());
+
+                        let mut output_buf = Vec::with_capacity(events.len());
                         let mut buf = Vec::with_capacity(4); // also an arbitrary,
                                                              // smallish constant
                         for v in events {
                             t.transform(&mut buf, v);
-                            output.append(&mut buf);
+                            output_buf.append(&mut buf);
                         }
                         emit!(&EventsSent {
-                            count: output.len(),
-                            byte_size: output.size_of(),
+                            count: output_buf.len(),
+                            byte_size: output_buf.size_of(),
                         });
```

Comment:
I think this may need to go after the `output.send_all` below. This stream conversion makes it easier to do that now.